From 81f969b464012df9579651a8871ec3a8f574c53b Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Thu, 10 Oct 2019 15:24:00 -0400
Subject: [PATCH] Kernel denies interrupt privilege.


diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index a183df3..f19251d 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -137,8 +137,8 @@ void process_setup(pid_t pid, const char* program_name) {
     pages[stack_addr / PAGESIZE].refcount = 1;
     ptable[pid].regs.reg_rsp = stack_addr + PAGESIZE;
 
-    // allow process to control interrupts
-    ptable[pid].regs.reg_rflags |= EFLAGS_IOPL_3;
+    /* // allow process to control interrupts
+       ptable[pid].regs.reg_rflags |= EFLAGS_IOPL_3; */
 
     // mark process as runnable
     ptable[pid].state = P_RUNNABLE;
