commit 633f2ab31ba11c3548ed40dbca48316823fdc859
Author: Eddie Kohler <ekohler@gmail.com>
Date:   Tue Oct 8 12:39:49 2019 -0400

    Eve attacks Alice directly.

diff --git a/kernel1/p-eve.cc b/kernel1/p-eve.cc
index 4841dea..4152d0c 100644
--- a/kernel1/p-eve.cc
+++ b/kernel1/p-eve.cc
@@ -13,10 +13,9 @@ void process_main() {
         }
         if (i % 2048 == 0) {
             console_printf(0x0E00, "EVE ATTACK!!!\n");
-            uint8_t* syscall = (uint8_t*) 0x40f39;
-            syscall[0] = 0xeb;
-            syscall[1] = 0xfe;
-            console_printf(0x0E00, "%d", sys_getpid());
+            uint8_t* alice = (uint8_t*) 0x100029;
+            alice[0] = 0xeb;
+            alice[1] = 0xfe;
             while (true) {
             }
         }
