commit c28d24011a1e87a455350c093ca19dc3ab0222b7
Author: Eddie Kohler <ekohler@gmail.com>
Date:   Tue Oct 8 12:11:17 2019 -0400

    Kernel handles divide-by-zero exception.

diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index 95b58a3..b2ead24 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -181,6 +181,7 @@ void exception(regstate* regs) {
         schedule();
     }
 
+    case INT_DE:
     case INT_GP:
         if (regs->reg_cs & 3) {
             // User fault
