From e01f19d8d6d688a0ce8476ac90c5cf3c74d40cb5 Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Thu, 10 Oct 2019 15:25:54 -0400
Subject: [PATCH] Kernel protects its memory.


diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index 54c3307..0129c9e 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -67,8 +67,10 @@ void kernel(const char* command) {
     for (vmiter it(kernel_pagetable);
          it.va() < MEMSIZE_PHYSICAL;
          it += PAGESIZE) {
-        if (it.va() != 0) {
+        if (it.va() >= PROC_START_ADDR) {
             it.map(it.va(), PTE_P | PTE_W | PTE_U);
+        } else if (it.va() != 0) {
+            it.map(it.va(), PTE_P | PTE_W);
         } else {
             it.map(it.va(), 0);
         }
