commit 55260ebb7a2794945aeb1513f2c1b86b870e15b5
Author: Eddie Kohler <ekohler@gmail.com>
Date:   Tue Oct 8 12:07:30 2019 -0400

    Kernel denies interrupt privilege.

diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index 28d01ff..3c8d226 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -137,8 +137,8 @@ void process_setup(pid_t pid, const char* program_name) {
     pages[stack_addr / PAGESIZE].refcount = 1;
     ptable[pid].regs.reg_rsp = stack_addr + PAGESIZE;
 
-    // allow process to control interrupts
-    ptable[pid].regs.reg_rflags |= EFLAGS_IOPL_3;
+    /* // allow process to control interrupts
+       ptable[pid].regs.reg_rflags |= EFLAGS_IOPL_3; */
 
     // mark process as runnable
     ptable[pid].state = P_RUNNABLE;
