From 54d2d1035dbfb6dc2f200b3aa477e9f963a719a8 Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Thu, 10 Oct 2019 15:23:44 -0400
Subject: [PATCH] Disable interrupts.


diff --git a/kernel1/p-eve.cc b/kernel1/p-eve.cc
index 4230c43..500743b 100644
--- a/kernel1/p-eve.cc
+++ b/kernel1/p-eve.cc
@@ -13,6 +13,7 @@ void process_main() {
         }
         if (i % 2048 == 0) {
             console_printf(0x0E00, "EVE ATTACK!!!\n");
+            asm volatile("cli");
             while (true) {
             }
         }
