commit 7bf668f0afc0b06d79909e0afadb0ce83e180c0e
Author: Eddie Kohler <ekohler@gmail.com>
Date:   Tue Oct 8 12:08:56 2019 -0400

    Kernel handles GPF.

diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index 3c8d226..95b58a3 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -181,6 +181,15 @@ void exception(regstate* regs) {
         schedule();
     }
 
+    case INT_GP:
+        if (regs->reg_cs & 3) {
+            // User fault
+            current->state = P_BROKEN;
+            break;
+        } else {
+            goto unhandled;
+        }
+
     case INT_PF: {
         // Analyze faulting address and access type.
         uintptr_t addr = rdcr2();
