From ada0b3c0a566831068e8a354c58243243aafd21e Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Thu, 10 Oct 2019 15:24:11 -0400
Subject: [PATCH] Kernel handles GPF.


diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index f19251d..09b964a 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -182,6 +182,15 @@ void exception(regstate* regs) {
         schedule();
     }
 
+    case INT_GP:
+        if (regs->reg_cs & 3) {
+            // User fault
+            current->state = P_BROKEN;
+            break;
+        } else {
+            goto unhandled;
+        }
+
     case INT_PF: {
         // Analyze faulting address and access type.
         uintptr_t addr = rdcr2();
