commit b08076a6878b7e38e8efbf7a03857463410088c0
Author: Eddie Kohler <ekohler@gmail.com>
Date:   Tue Oct 8 12:06:39 2019 -0400

    Disable interrupts.

diff --git a/kernel1/p-eve.cc b/kernel1/p-eve.cc
index 4230c43..500743b 100644
--- a/kernel1/p-eve.cc
+++ b/kernel1/p-eve.cc
@@ -13,6 +13,7 @@ void process_main() {
         }
         if (i % 2048 == 0) {
             console_printf(0x0E00, "EVE ATTACK!!!\n");
+            asm volatile("cli");
             while (true) {
             }
         }
