commit c8a6c9ed4038d29f320712f7237cbe2d189e8bf8
Author: Eddie Kohler <ekohler@gmail.com>
Date:   Thu Oct 10 12:46:10 2019 -0400

    Fix pagefault handling.

diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index 4b888aa..1acdcac 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -67,7 +67,8 @@ void kernel(const char* command) {
     for (vmiter it(kernel_pagetable);
          it.va() < MEMSIZE_PHYSICAL;
          it += PAGESIZE) {
-        if (it.va() >= PROC_START_ADDR) {
+        if (it.va() >= PROC_START_ADDR
+            || it.va() == CONSOLE_ADDR) {
             it.map(it.va(), PTE_P | PTE_W | PTE_U);
         } else if (it.va() != 0) {
             it.map(it.va(), PTE_P | PTE_W);
@@ -206,7 +207,12 @@ void exception(regstate* regs) {
         console_printf(COLOR_ERROR,
               "%s %d page fault on %p (%s %s, rip=%p)!\n",
               entity, current->pid, addr, operation, problem, regs->reg_rip);
-        goto unhandled;
+        if (regs->reg_errcode & PFERR_USER) {
+            current->state = P_BROKEN;
+            break;
+        } else {
+            goto unhandled;
+        }
     }
 
     default:
