From 42b2b0366e60f2db8bbe1b72e8cff3e8eedb9a3a Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Thu, 10 Oct 2019 15:24:53 -0400
Subject: [PATCH] Kernel handles divide-by-zero exception.


diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index 09b964a..54c3307 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -182,6 +182,7 @@ void exception(regstate* regs) {
         schedule();
     }
 
+    case INT_DE:
     case INT_GP:
         if (regs->reg_cs & 3) {
             // User fault
