commit 35b4c4dff4f6fab881d92266da951846af195ae6
Author: Eddie Kohler <ekohler@gmail.com>
Date:   Tue Oct 8 12:05:53 2019 -0400

    Enable timer interrupt.

diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index c56aa61..28d01ff 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -56,6 +56,9 @@ void kernel(const char* command) {
     // initialize hardware
     init_hardware();
 
+    // set up timer interrupt every 0.001sec
+    init_timer(1000);
+
     // clear screen
     console_clear();
 
@@ -173,6 +176,11 @@ void exception(regstate* regs) {
     // Actually handle the exception.
     switch (regs->reg_intno) {
 
+    case INT_IRQ + IRQ_TIMER: {
+        lapicstate::get().ack();
+        schedule();
+    }
+
     case INT_PF: {
         // Analyze faulting address and access type.
         uintptr_t addr = rdcr2();
