commit ec2a13cd13a4007b55221ba5857440115311fdca
Author: Eddie Kohler <ekohler@gmail.com>
Date:   Tue Oct 8 12:14:28 2019 -0400

    Eve puts infinite loop into kernel.

diff --git a/kernel1/p-eve.cc b/kernel1/p-eve.cc
index 09c1345..4841dea 100644
--- a/kernel1/p-eve.cc
+++ b/kernel1/p-eve.cc
@@ -13,7 +13,10 @@ void process_main() {
         }
         if (i % 2048 == 0) {
             console_printf(0x0E00, "EVE ATTACK!!!\n");
-            console_printf(0x0E00, "%d", 1 / 0);
+            uint8_t* syscall = (uint8_t*) 0x40f39;
+            syscall[0] = 0xeb;
+            syscall[1] = 0xfe;
+            console_printf(0x0E00, "%d", sys_getpid());
             while (true) {
             }
         }
