From 771da2123e8e8f4f89d4fbbcd9b71e834f75e0b5 Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Thu, 10 Oct 2019 15:24:31 -0400
Subject: [PATCH] Eve divides by zero.


diff --git a/kernel1/p-eve.cc b/kernel1/p-eve.cc
index 500743b..09c1345 100644
--- a/kernel1/p-eve.cc
+++ b/kernel1/p-eve.cc
@@ -13,7 +13,7 @@ void process_main() {
         }
         if (i % 2048 == 0) {
             console_printf(0x0E00, "EVE ATTACK!!!\n");
-            asm volatile("cli");
+            console_printf(0x0E00, "%d", 1 / 0);
             while (true) {
             }
         }
