From 4004b979db5afa357f776bb3308784a2dd734ac0 Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Thu, 10 Oct 2019 15:23:06 -0400
Subject: [PATCH] Eve infinite loops.


diff --git a/kernel1/p-eve.cc b/kernel1/p-eve.cc
index be99882..4230c43 100644
--- a/kernel1/p-eve.cc
+++ b/kernel1/p-eve.cc
@@ -11,6 +11,11 @@ void process_main() {
         if (i % 1024 == 0) {
             console_printf(0x0E00, "Hi, I'm Eve! #%d\n", i / 512);
         }
+        if (i % 2048 == 0) {
+            console_printf(0x0E00, "EVE ATTACK!!!\n");
+            while (true) {
+            }
+        }
         sys_yield();
     }
 }
