From 7fad6c7b1f402f9450da44126d209d50a4672ff2 Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Thu, 10 Oct 2019 15:23:29 -0400
Subject: [PATCH] Enable timer interrupt.


diff --git a/kernel1/kernel.cc b/kernel1/kernel.cc
index b46cc3c..a183df3 100644
--- a/kernel1/kernel.cc
+++ b/kernel1/kernel.cc
@@ -56,6 +56,9 @@ void kernel(const char* command) {
     // initialize hardware
     init_hardware();
 
+    // set up timer interrupt every 0.001sec
+    init_timer(1000);
+
     // clear screen
     console_clear();
 
@@ -174,6 +177,11 @@ void exception(regstate* regs) {
     // Actually handle the exception.
     switch (regs->reg_intno) {
 
+    case INT_IRQ + IRQ_TIMER: {
+        lapicstate::get().ack();
+        schedule();
+    }
+
     case INT_PF: {
         // Analyze faulting address and access type.
         uintptr_t addr = rdcr2();
